### ISSUE TITLE
Notifications about stream updates not working in some cases

### ISSUE THREAD

**Issue by yurikuzn** [espocrm/espocrm#1033]

**Comment by Sloff** [comment]
Will this also resolve the issue where users are unable to see certain stream messages? For example they are unable to see any email sent messages in the stream, but I've confirmed that they exist

**Comment by yurikuzn** [comment]
Not sure. Does the use has an access to see those emails?

**Comment by Sloff** [comment]
Yes, also they user can see the emails in the "History" panel, but not the stream

**Comment by yurikuzn** [comment]
Are those email sent before upgrade or after?

**Comment by Sloff** [comment]
Before, I'm not sure if emails sent after are affected

**Comment by yurikuzn** [comment]
Unfortunately, old emails won't be seen in stream if user has team or own read access level.

**Comment by Sloff** [comment]
Is this with regards to the Stream access value?

**Comment by yurikuzn** [comment]
No. Read access.

**Comment by Sloff** [comment]
Read access of Emails or Cases? (This is occurring on the stream of Cases)

**Comment by yurikuzn** [comment]
emails.

**Comment by Sloff** [comment]
Could you point me to the code that enforces this? I can't give all the users full read permission of emails because of security issues, but they also have to see the content in the stream in order to print it for Audit reasons

**Comment by yurikuzn** [comment]
Future email should be listed fine.

This acl check is very tricky:
https://github.com/espocrm/espocrm/blob/master/application/Espo/Services/Stream.php#L1338
https://github.com/espocrm/espocrm/blob/master/application/Espo/Services/Stream.php#L1358

**Comment by yurikuzn** [comment]
I'm going to write php script that will apply teams to existing note records.

**Comment by Sloff** [comment]
That would be awesome, thanks @yurikuzn

**Comment by yurikuzn** [comment]
Script will be automatically executed after upgrade to 5.4.2 in about 15 minutes by cron.
